src/components/workspace-utils.js

```diff
@@ -56,6 +58,31 @@ export const WorkspaceSelector = ({ workspaces, value, onChange, ...props }) =>
   })
 }
 
+export const SnapshotInfo = ({ snapshotId, snapshotName }) => {
+  const [snapshotInfo, setSelectedSnapshotInfo] = useState({})
+  const signal = Utils.useCancellation()
+
+  Utils.useOnMount(() => {
+    const loadSnapshotInfo = async () => {
+      const snapshotInfo = await Ajax(signal).DataRepo.snapshot(snapshotId).details()
+      setSelectedSnapshotInfo(snapshotInfo)
+    }
+
+    loadSnapshotInfo()
+  })
+
+  const { name, description, createdDate } = snapshotInfo
+
+  return _.isEqual(snapshotInfo, {}) ? spinnerOverlay : div({
```

Comment:
We prefer leaving something that hasn't yet been loaded undefined:
```suggestion
  const [snapshotInfo, setSelectedSnapshotInfo] = useState()
  const signal = Utils.useCancellation()

  Utils.useOnMount(() => {
    const loadSnapshotInfo = async () => {
      const snapshotInfo = await Ajax(signal).DataRepo.snapshot(snapshotId).details()
      setSelectedSnapshotInfo(snapshotInfo)
    }

    loadSnapshotInfo()
  })

  const { name, description, createdDate } = snapshotInfo || {}

  return snapshotInfo === undefined ? spinnerOverlay : div({
```